multi_map_it_test.go

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package hazelcast_test
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	hz "github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+func TestMultiMap_Put(t *testing.T) {
+	it.MultiMapTester(t, func(t *testing.T, m *hz.MultiMap) {
+		targetValue := "value1"
+		assert.False(t, !it.MustBool(m.Put(context.Background(), "key", targetValue)), "multi-map put failed")
+		tmp := it.MustValue(m.Get(context.Background(), "key"))
+		it.AssertEquals(t, []interface{}{targetValue}, tmp)
+
+		targetValue2 := "value2"
+		assert.False(t, !it.MustBool(m.Put(context.Background(), "key", targetValue2)), "multi-map put failed")
+		tmp = it.MustValue(m.Get(context.Background(), "key"))
+		assert.ElementsMatch(t, []interface{}{targetValue, targetValue2}, tmp)
+	})
+}
+
+//todo check if this putAll api addresses the need
+func TestMultiMap_PutAll(t *testing.T) {
+	it.MultiMapTester(t, func(t *testing.T, m *hz.MultiMap) {
+		existingValue := "v4"
+		assert.False(t, !it.MustBool(m.Put(context.Background(), "k1", existingValue)), "multimap put failed")
+		value := it.MustValue(m.Get(context.Background(), "k1"))
+		it.AssertEquals(t, []interface{}{existingValue}, value)
+		it.Must(m.PutAll(context.Background(), "k1", "v1", "v2", "v3"))
```

Comment:
The same comments as the test above, about separating the operation to its own line. It would be great if you could update other tests as well.

Reply:
Fixed as suggested